File: Second/SecondProject/Plant-entity classes/Tree.cpp
//------------------------------------------------------------------------------
// Tree.cpp - provides implementation for the tree-struct
// and functions working with it.
//------------------------------------------------------------------------------
#include "Tree.h"

// Default "tree"-object constructor.
Tree::Tree() : Plant(){
    this->age_ = 0;
}

/* Special constructor, which has different ways of creating a
 * "tree"-object depending on what type of input was selected. */
Tree::Tree(bool from_file, FILE *file) : Plant(from_file, file) {
    if (from_file) {
        fscanf(file, "%lld", &this->age_);
        this->age_ = abs(this->age_);
    } else {
        this->age_ = rand() % 500 + 1;
    }
}

// Sets the C-string-type data of a given tree into a char**-pointer.
char *Tree::toString() {
    char *plant_data = static_cast<Plant>(*this).toString();
    char *data = new char[101];
    data[100] = '\0';
    int print_count = snprintf(data, 100, "Tree %s:%lld years", plant_data, this->age_);
    data[print_count] = '\0';
    delete [] plant_data;
    return data;
}

File: Third/ThirdProject/plants/Tree.py
# ------------------------------------------------------------------------------
# Tree.py - provides declaration and implementation for the tree-class
# and functions working with it.
# ------------------------------------------------------------------------------
from plants.Plant import *


class Tree(Plant):

    def __init__(self, is_random: bool, params: list) -> None:
        """
        Special constructor, which has different ways of creating a
        "tree"-object depending on what type of input was selected.

        :param is_random: Logical parameter, specifying a method of extracting input data -
        random generation or file-input.

        :param params: A list of params passed in case of file-input.
        """
        if not is_random and params.__len__() < 3:
            self = 0
            return
        super().__init__(is_random, params)
        self.age_ = random.randint(0, 500) if is_random else abs(int(params[2])) % 501

    def toString(self) -> str:
        """
        A function to get string-type presentation of tree-data.

        :return: A string, containing information about a tree.
        """
        return f"Tree {super().toString()}: {self.age_} years old"
